src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -0,0 +1,151 @@
+
+// Copyright 2020 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "kafka/server/handlers/alter_partition_reassignments.h"
+
+#include "cluster/metadata_cache.h"
+#include "cluster/topics_frontend.h"
+#include "cluster/types.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/protocol/schemata/alter_partition_reassignments_request.h"
+#include "kafka/protocol/schemata/alter_partition_reassignments_response.h"
+#include "kafka/server/errors.h"
+#include "kafka/server/fwd.h"
+#include "kafka/server/quota_manager.h"
+#include "model/errc.h"
+#include "model/metadata.h"
+#include "model/namespace.h"
+#include "model/timeout_clock.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/sstring.hh>
+
+#include <absl/container/node_hash_map.h>
+
+#include <algorithm>
+#include <chrono>
+#include <iterator>
+#include <system_error>
+#include <vector>
+
+namespace kafka {
+
+error_code error_code_as_kafka_error(std::error_code ec) {
+    switch (static_cast<model::errc>(ec.value())) {
+    case model::errc::success:
+        return error_code::none;
+    case model::errc::topic_name_len_exceeded:
+    case model::errc::topic_name_empty:
+    case model::errc::invalid_topic_name:
+    case model::errc::forbidden_topic_name:
+        return error_code::invalid_topic_exception;
+    default:
+        return error_code::unknown_server_error;
+    }
+}
+
+template<>
+ss::future<response_ptr> alter_partition_reassignments_handler::handle(
+  request_context ctx, [[maybe_unused]] ss::smp_service_group ssg) {
+    alter_partition_reassignments_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+    alter_partition_reassignments_response resp;
+
+    if (request.data.topics.empty()) {
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    resp.data.responses.reserve(request.data.topics.size());
+    auto timeout_ms = request.data.timeout_ms;
+    kafka::error_code top_level_err{kafka::error_code::none};
+
+    for (const reassignable_topic& topic : request.data.topics) {
+        reassignable_topic_response topic_response{.name = topic.name};
+
+        for (const reassignable_partition& partition : topic.partitions) {
+            model::ntp ntp{
+              model::ns{"kafka"}, topic.name, partition.partition_index};
```

Comment:
Hardcoding for the kafka namespaces makes sense to me here because the request will always be for partitions in a topic and thus always in the kafka namespace. I'm open to ideas though.

Reply:
using kafka namespace here makes sense. can you use the constants defined in model/namespace.h?